packages/server/test/index.ts

```diff
@@ -39,6 +37,18 @@ async function createStoreController () {
   })
 }
 
+async function testErrorFormat (t: Test, err: GotError, expectedCode: number) {
+  if (err instanceof HTTPError) {
+    t.equal(err.statusCode, expectedCode, `response code should be a ${expectedCode}`)
+  }
+  try {
+    const responseBody = JSON.parse(err.response.body)
+    t.ok(responseBody.error, 'error field should be set in response body')
+  } catch (err) {
+    t.error(err)
+  }
```

Comment:
you can simplify it to

```ts
t.ok(JSON.parse(err.response.body).error, 'error field should be set in response body')
```

if parsing the JSON will fail then the test will fail, so that's ok

Reply:
It seems like this doesn't actually work. The follow error shows up in the console when `JSON.parse` fails and all tests stop running (the script hangs):

```
(node:1612) UnhandledPromiseRejectionWarning: SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at Object.<anonymous> (/mnt/d/tejun/Github/pnpm/pnpm/packages/server/test/index.ts:400:15)
    at Generator.throw (<anonymous>)
    at rejected (/mnt/d/tejun/Github/pnpm/pnpm/packages/server/test/index.ts:5:65)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:229:7)
(node:1612) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:1612) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future,
promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Any easy way to fix this while still keeping it clean?

Reply:
My guess is that `JSON.parse` is throwing an exception, which gets propagated up the entire function instead of just failing the test. This causes the test to exit without calling `t.end()`